<a href="https://colab.research.google.com/github/tejaswi1995prakash/COVID19india-report/blob/master/India_COVID19_Report.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **COVID19 INDIA REPORT**


In [0]:
import pandas as pd
import numpy as np


## **Problem Statement**
This analysis is to understand the cases in India and the impact that it has caused. While also understanding Lockdown and their effects.

In [0]:
raw_data = pd.read_csv('https://api.covid19india.org/csv/latest/raw_data.csv')
raw_data.head()

Patient Number  ...                                Backup Notes
0               1  ...                          Student from Wuhan
1               2  ...                          Student from Wuhan
2               3  ...                          Student from Wuhan
3               4  ...         Travel history to Italy and Austria
4               5  ...  Travel history to Dubai, Singapore contact

[5 rows x 20 columns]

# **Initial set of Patients**

In [0]:
"India got it's first case on {x}. The first case was in {city} city in {state}".format(x = raw_data.iloc[0]['Date Announced'],city = raw_data.iloc[0]['Detected City'], state =raw_data.iloc[0]['Detected State'] )

"India got it's first case on 30/01/2020. The first case was in Thrissur city in Kerala"

Here is the state wise report of the cases in first month

In [0]:
#raw_data.loc[:,['Date Announced']] = pd.to_datetime(raw_data.loc[:,['Date Announced']],format = '%d%m%Y')

raw_data['Date Announced'] = pd.to_datetime(raw_data['Date Announced'] ,format = '%d/%m/%Y')

In [0]:
raw_data.sort_values(['Date Announced']).head(200).groupby('Detected State')[['Patient Number']].count().sort_values('Patient Number', ascending = False).reset_index()

Detected State  Patient Number
0          Tamil Nadu              31
1           Telangana              30
2               Delhi              29
3              Kerala              24
4      Andhra Pradesh              15
5         Maharashtra              15
6               Assam              11
7      Madhya Pradesh              11
8           Rajasthan              11
9   Jammu and Kashmir               7
10            Gujarat               5
11          Karnataka               5
12         Chandigarh               2
13      Uttar Pradesh               2
14              Bihar               1
15             Odisha               1

Observation : Maharashtra is the state with the highest number of cases in the first month.
This may be because of the fact that it is densely populated 

## **People responsible for most spreads**

In [0]:
raw_data.groupby(['Contracted from which Patient (Suspected)'])[['Patient Number']].count().sort_values(by = 'Patient Number',ascending = False)[:10]

Patient Number
Contracted from which Patient (Suspected)                
E0                                                    958
P689                                                   42
P20410                                                 20
P182                                                   20
P6                                                     16
P301                                                   13
P8460                                                  12
P2868                                                  11
E1                                                     11
P764                                                   11

We see that E0 alone is responsible for spreading to 958 people.

In [0]:
raw_data.query('`Contracted from which Patient (Suspected)` == "E0"').head(10)

Patient Number  ...                                       Backup Notes
140             141  ...  Indonesian national; The patient had flown int...
162             163  ...                                         Indonesian
163             164  ...                                         Indonesian
164             165  ...                                         Indonesian
165             166  ...                                         Indonesian
166             167  ...                                         Indonesian
167             168  ...                                         Indonesian
168             169  ...                                         Indonesian
250             252  ...                                 Indonesian citizen
251             253  ...                                 Indonesian citizen

[10 rows x 20 columns]

It looks like the patient E0 is related to Tablighi Jamat Conference in Delhi

In [0]:
statewise = pd.read_csv('https://api.covid19india.org/csv/latest/state_wise_daily.csv')

In [0]:
statewise_test = pd.read_csv('https://api.covid19india.org/csv/latest/statewise_tested_numbers_data.csv')

In [0]:
statewise['Date'] = pd.to_datetime(statewise['Date'])

In [0]:
states_only = statewise.drop(columns='TT')


In [0]:
s1 = pd.melt(states_only,id_vars=['Date','Status']).sort_values(['Date','Status','variable'])
s1.head()


Date     Status variable  value
0   2020-03-14  Confirmed       AN    0.0
150 2020-03-14  Confirmed       AP    1.0
300 2020-03-14  Confirmed       AR    0.0
450 2020-03-14  Confirmed       AS    0.0
600 2020-03-14  Confirmed       BR    0.0

In [0]:
ss = pd.crosstab(s1['variable'],s1['Status'],values= s1['value'],aggfunc= np.sum)
ss.reset_index()
ss["Mortality_Rate"]=(ss['Deceased']/ss['Confirmed'])
ss.sort_values('Mortality_Rate',ascending=False)


Status    Confirmed  Deceased  Recovered  Mortality_Rate
variable                                                
ML             12.0       1.0       10.0        0.083333
MP           2788.0     152.0      624.0        0.054519
WB            922.0      48.0      151.0        0.052061
GJ           5054.0     262.0      896.0        0.051840
HP             40.0       2.0       34.0        0.050000
MH          12296.0     520.0     2000.0        0.042290
KA            601.0      25.0      271.0        0.041597
TG           1061.0      29.0      499.0        0.027333
JH            115.0       3.0       22.0        0.026087
PB            772.0      20.0      112.0        0.025907
RJ           2772.0      68.0     1242.0        0.024531
AS             43.0       1.0       33.0        0.023256
AP           1525.0      33.0      441.0        0.021639
UP           2487.0      43.0      698.0        0.017290
UT             59.0       1.0       39.0        0.016949
DL           4122.0      64.0     1256.0        0.015526
HR            376.0       5.0      241.0        0.013298
JK            651.0       8.0      254.0        0.012289
TN           2757.0      29.0     1341.0        0.010519
BR            481.0       4.0      117.0        0.008316
KL            500.0       4.0      400.0        0.008000
OR            160.0       1.0       56.0        0.006250
PY             12.0       0.0        5.0        0.000000
TR              4.0       0.0        2.0        0.000000
AN             33.0       0.0       26.0        0.000000
MZ              1.0       0.0        0.0        0.000000
MN              2.0       0.0        2.0        0.000000
GA              7.0       0.0        7.0        0.000000
CT             43.0       0.0       36.0        0.000000
CH             94.0       0.0       19.0        0.000000
AR              1.0       0.0        1.0        0.000000
LA             23.0       0.0       17.0        0.000000
DD              0.0       0.0        0.0             NaN
DN              0.0       0.0        0.0             NaN
LD              0.0       0.0        0.0             NaN
NL              0.0       0.0        0.0             NaN
SK              0.0       0.0        0.0             NaN

We can now see the total number of Confirmed,Deceased and Recovered cases state wise.
Though Meghalaya has the highest Mortality rate, It should be noted that the sample size is very less in Meghalaya. Madhya Pradesh on the other hand has a mortality rate around 5.5% which is followed by West Bengal which is around 5.2%

# **Visualization**
# Statewise representation 

In [0]:

import plotly.graph_objects as go

fig = go.Figure([go.Bar(x=s1['variable'].unique(), y=ss['Confirmed'])])
fig.show()

In [0]:
import plotly.graph_objects as go
#states = ([s1['variable'].unique()])


fig = go.Figure()
fig.add_trace(go.Bar(x=s1['variable'].unique(),
                y=ss['Confirmed'],
                name='Total Cases',
                marker_color='rgb(15, 83, 109)'
                ))
fig.add_trace(go.Bar(x=s1['variable'].unique(),
                y=ss['Deceased'],
                name='Deaths',
                marker_color='rgb(26, 118, 255)'
                ))
fig.update_layout(
    title='COVID19 INDIA',
    xaxis_tickfont_size=14,
    yaxis=dict(
        title='Number of cases',
        titlefont_size=16,
        tickfont_size=14,
        ),

    legend=dict(
        x=0,
        y=1.0,
        bgcolor='rgba(250, 255, 255, 0)',
        bordercolor='rgba(255, 255, 255, 0)'
    ),

  barmode='group',
    bargap=0.15, # gap between bars of adjacent location coordinates.
    bargroupgap=0.1 # gap between bars of the same location coordinate.
)
fig.show()


# **Line Graph**